### PR TITLE
[MI-2361] Updated API to create/delete "Run" event subscriptions with…

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
… filters

### CODE REVIEW

File: server/store/subscriptions.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (subscriptionList *SubscriptionList) AddSubscription(userID string, subscri
...
+	subscriptionKey := GetSubscriptionKey(userID, subscription.ProjectName, subscription.ChannelID, subscription.EventType, subscription.Repository, subscription.TargetBranch, subscription.PullRequestCreatedBy, subscription.PullRequestReviewersContains, subscription.PushedBy, subscription.MergeResult, subscription.NotificationType, subscription.AreaPath, subscription.ReleasePipeline, subscription.BuildPipeline, subscription.BuildStatus, subscription.ApprovalType, subscription.ApprovalStatus, subscription.StageName, subscription.ReleaseStatus, subscription.RunPipeline, subscription.RunStageName, subscription.RunEnvironmentName, subscription.RunStageNameID, subscription.RunStageStateID, subscription.RunStageResultID, subscription.RunStateID, subscription.RunResultID)
```
Comment (Kshitij-Katiyar) [inline server/store/subscriptions.go:61]: make this multiline its very long

File: server/store/subscriptions.go [hunk 2]
```diff
@@ -146,7 +154,7 @@ func deleteSubscriptionAtomicModify(subscription *serializers.SubscriptionDetail
...
+	subscriptionKey := GetSubscriptionKey(subscription.MattermostUserID, subscription.ProjectName, subscription.ChannelID, subscription.EventType, subscription.Repository, subscription.TargetBranch, subscription.PullRequestCreatedBy, subscription.PullRequestReviewersContains, subscription.PushedBy, subscription.MergeResult, subscription.NotificationType, subscription.AreaPath, subscription.ReleasePipeline, subscription.BuildPipeline, subscription.BuildStatus, subscription.ApprovalType, subscription.ApprovalStatus, subscription.StageName, subscription.ReleaseStatus, subscription.RunPipeline, subscription.RunStageName, subscription.RunEnvironmentName, subscription.RunStageNameID, subscription.RunStageStateID, subscription.RunStageResultID, subscription.RunStateID, subscription.RunResultID)
```
Comment (Kshitij-Katiyar) [inline server/store/subscriptions.go:157]: same

File: server/store/utils.go [hunk 3]
```diff
@@ -136,9 +136,9 @@ func GetSubscriptionListMapKey() string {
 	return GetKeyHash(constants.SubscriptionPrefix)
 }
 
-func GetSubscriptionKey(mattermostUserID, projectID, channelID, eventType, repository, targetBranch, pullrequestCreatedBy, pullRequestReviewersContains, pushedBy, mergeResult, notificationType, areaPath, releasePipeline, buildPipeline, buildStatus, approvalType, approvalStatus, stageName, releaseStatus string) string {
+func GetSubscriptionKey(mattermostUserID, projectID, channelID, eventType, repository, targetBranch, pullrequestCreatedBy, pullRequestReviewersContains, pushedBy, mergeResult, notificationType, areaPath, releasePipeline, buildPipeline, buildStatus, approvalType, approvalStatus, stageName, releaseStatus, runPipeline, runStageName, runEnvironmentName, runStageNameID, runStageStateID, runStageResultID, runStateID, runResultID string) string {
 	// TODO: Optimize subscription key
```
Comment (Kshitij-Katiyar) [inline server/store/utils.go:140]: is this todo comment fixed

**Review by ayusht2810 — APPROVED** [review]

File: server/store/utils.go (hunk 3 above)
Comment (raghavaggarwal2308) [reply server/store/utils.go:140]: Yes, its fixed will sync the PRs

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**